tests/reward_claims_e2e.rs

```diff
@@ -0,0 +1,515 @@
+                let view = leaf_info.leaf.view_number();
+
+                if height > latest_height {
+                    latest_height = height;
+                    println!(
+                        "Block decided - Height: {}, View: {}, Target: {}",
+                        height,
+                        view.u64(),
+                        target_block
+                    );
+                }
+
+                if height >= target_block {
+                    println!(
+                        "Reached target block height {}, view number: {}, 3 epochs have elapsed",
+                        height,
+                        view.u64()
+                    );
+                    break;
+                }
+            }
+
+            if latest_height >= target_block {
+                break;
+            }
+        }
+    }
+
+    // Now run the prover once to generate and submit a proof for the current state
+    println!("Running prover once to generate proof...");
+    run_prover_once(prover_config, SequencerApiVersion::instance()).await?;
+    println!("Prover completed successfully");
+
+    // Get the finalized state from the Light Client contract
+    println!("Getting finalized state from Light Client contract...");
+    let light_client_contract = LightClientV3::new(lc_proxy_addr, &provider);
+    let finalized_state = light_client_contract.finalizedState().call().await?;
+    let auth_root = light_client_contract.authRoot().call().await?;
+
+    println!(
+        "Light Client finalized state - Block height: {}, View: {}, Auth root: {:#x}",
+        finalized_state.blockHeight, finalized_state.viewNum, auth_root._0
+    );
+
+    // Use the block height from the light client contract for the reward proof
+    let lc_block_height = finalized_state.blockHeight;
+    let lc_view_number = finalized_state.viewNum;
+
+    println!("Reward claims E2E test setup completed successfully!");
+    println!("Contract addresses:");
+    println!("  Light Client proxy: {}", lc_proxy_addr);
+    println!("  Fee Contract proxy: {}", fee_proxy_addr);
+    println!("  ESP Token proxy: {}", token_proxy_addr);
+    println!("  Stake Table proxy: {}", stake_table_proxy_addr);
+    println!("  RewardClaim proxy: {}", reward_claim_proxy_addr);
+    println!("Network info:");
+    println!("  L1 URL: {}", l1_url);
+    println!("  Sequencer API port: {}", sequencer_api_port);
+    println!("  Relay server port: {}", relay_server_port);
+
+    let sequencer_url: Url = format!("http://localhost:{sequencer_api_port}")
+        .parse()
+        .unwrap();
+    println!(
+        "Fetching reward proof for LC block height: {}, LC view number: {}",
+        lc_block_height, lc_view_number
+    );
+
+    let reward_account = format!("0x{:x}", claimer_address);
+    println!(
+        "Testing reward claim for validator account: {}",
+        reward_account
+    );
+
+    let reward_proof_url = format!(
+        "{}catchup/{}/{}/reward-account-v2/{}",
```

Comment:
This will be changed to use the reward API but we have to first change the return type of that API to something more friendly to convert into EVM compatible types.